CMakeLists.txt

```diff
@@ -542,24 +566,32 @@ endif(USE_MPI)
 ###############################################################################
 # Find BLAS library for CPU compilation
 if(COMPILE_CPU)
-  set(EXT_LIBS ${EXT_LIBS} intgemm)       # Move the intgemm bits on top since they compile with every single variant
+  if(USE_INTGEMM)
+      set(EXT_LIBS ${EXT_LIBS} intgemm)       # Move the intgemm bits on top since they compile with every single variant
+  endif(USE_INTGEMM)
+
+  if(USE_RUY)
+    set(EXT_LIBS ${EXT_LIBS} ruy)
+  endif(USE_RUY)
+
   add_definitions(-DCOMPILE_CPU=1)        # Move the compile CPU definition on top since we want to compile intgemm when we set compile CPU
                                           # in case a BLAS vendor is not found, we have a runtime error, although we should probably not allow the compilation to go on
                                           # if there are BLAS vendors, we have other runtime checks with sane error messages.
-  if(USE_WASM_COMPATIBLE_BLAS)
+  if(USE_ONNX_SGEMM)
     ## Use a wasm compatible BLAS
+    ## ^ SGEMM != BLAS
     set(EXT_LIBS ${EXT_LIBS} onnx-sgemm)
-    set(BLAS_FOUND TRUE)
-    set(BLAS_VENDOR "ONNX-SGEMM")
-    add_definitions(-DBLAS_FOUND=1  -DWASM_COMPATIBLE_BLAS=1) # Might be required in some cmake files further down the line, let's avoid using add_compile_definitions in this codeblock
-  elseif(APPLE AND USE_APPLE_ACCELERATE)
+    add_definitions(-DUSE_ONNX_SGEMM=1) # Might be required in some cmake files further down the line, let's avoid using add_compile_definitions in this codeblock
```

Comment:
ONNX_SGEMM is only necessary on non-apple. APPLE_ACCELERATE should be used on apple regardless of the arch.

Is this the current behaviour?

Reply:
The plan is to bring in some refactor in GEMM layer, so the flags can be flat than the current nested.
https://github.com/jerinphilip/sgemm

Reply:
Just saying that we have a full bells and whistles gemm on apple and we should make sure it's used.

Reply:
https://github.com/jerinphilip/sgemm/commit/8206c8e6bd0fca22ba60b8eec4747e493c115773